### ISSUE TITLE
Idea: Python script examples

### ISSUE THREAD

**Issue by raphaelbastide** [trufont/trufont#3]
Adding pre-existing basic commented python scripts could help efforts to dive into scripting. It could be a really nice pedagogic approach as well. What do you think?

**Comment by adrientetar** [comment]
```
from PyQt5.QtWidgets import QDialog, QPushButton, QVBoxLayout
import random

class FuzzDialog(QDialog):
    def __init__(self, parent=None):
        super(FuzzDialog, self).__init__(parent)
        layout = QVBoxLayout(self)
        fuzzBox = QPushButton("Fuzz", self)
        fuzzBox.clicked.connect(self.glyphFuzzer)
        layout.addWidget(fuzzBox)
        self.setLayout(layout)
    
    def glyphFuzzer(self):
        glyph = CurrentGlyph()
        if glyph is not None:
            for contour in glyph:
                for point in contour:
                    point.x += random.randrange(-4, 5)
                    point.y += random.randrange(-4, 5)
            glyph.dirty = True

FuzzDialog().exec_()
```

**Comment by adrientetar** [comment]
Note that I haven't set up a local event loop just yet for scripting so you have to use blocking widgets or they'll disappear. But that'll be fixed soon.

I think example scripts could live on a wiki or on the website. Some tutorial that walks you through examples could be a very good resource to have.

**Comment by raphaelbastide** [comment]
Your script example is exactly what I had in mind: Simple, with a direct graphic impact, and easily modifiable.

> I think example scripts could live on a wiki or on the website. 

So you don’t think embedding script into the program (just like [Processing does](http://i.stack.imgur.com/Ln0KE.png)) is a good idea? It can be helpful to work without Internet connection and give a better visibility to some basic scripted features (Stroke, Shadow, Add shape over).

**Comment by davelab6** [comment]
> I think example scripts could live on a wiki or on the website

I disagree, I think the core code tree (and including in the app bundle itself) is better; having the example scripts in a github repo provides wiki-like community contribution (via pull request) and having them on the github website provides code highlighting, search engine indexing, etc. 

It would be good for the app to be able to refresh the local copy from the web, similar to how text editors can load syntax highlighting schemas or plugins from the net

**Comment by adrientetar** [comment]
Alright, I just found how to allow non-blocking window creation w/o syntactic burden on the user.

Updated examples [here](https://gist.github.com/adrientetar/46368fd9ac693e6c3a4f).

**Comment by davelab6** [comment]
On 12 October 2015 at 08:04, Raphaël Bastide <notifications@github.com>
wrote:

> So you don’t think embedding script into the program (just like Processing
> does <http://i.stack.imgur.com/Ln0KE.png>) is a good idea? It can be
> helpful to work without Internet connection and give a better visibility to
> some basic scripted features (Stroke, Shadow, Add shape over).
>
I second Raphael on this - I think its super super valuable to have these
scripts both included/developed as part of the main codebase, and packaged
inside the program as 'batteries included' :)

Your latest example is awesome!!

**Comment by pathumego** [comment]
@adrientetar posted some scripts here. http://adrientetar.legtux.org/ I have a few recordings from the LGM workshop. I will edit it together and upload to youtube. I think this discussion should reopen about where to put the scripts. Probably a separate repo in the org? So it can be packaged separately and have its own issue tracker.
